testsuite/python/contact_time.py

```diff
@@ -0,0 +1,233 @@
+#
+# Copyright (C) 2025 The ESPResSo project
+#
+# This file is part of ESPResSo.
+#
+# ESPResSo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""
+Tests the Pairwise Distance observable and the ContactTimes accumulator.
+"""
+
+import unittest as ut
+import unittest_decorators as utx
+import numpy as np
+import espressomd
+import espressomd.accumulators
+import espressomd.observables
+
+
+class ContactTimeTest(ut.TestCase):
+
+    @classmethod
+    def calculate_minimum_image_distance(cls, coords1, coords2, size_box):
+        """
+        Calculates the minimum distance between two particles in the simulation box,
+        taking into account the minimum image convention and periodic boundary conditions.
+
+        Args:
+            coords1(`list`): array with the cartesian coordinates of the first particle.
+            coords2(`list`): array with the cartesian coordinates of the second particle.
+            size_box(`list`): xyz size of the simulation box.
+
+        Returns:
+            dist(`float`): minimum distance between the particles
+        """
+        # Sanity check, check that all input variables have the same size
+        assert len(coords1) == len(size_box)
+        assert len(coords2) == len(size_box)
+
+        n_axes = len(size_box)
+        coords = {"particle1": coords1,
+                  "particle2": coords2}
+
+        # Fold particles
+        for particle in coords.keys():
+            for axis in range(n_axes):
+                coords[particle][axis] = coords[particle][axis] % size_box[axis]
+
+        # Calculate minimum distance
+        dist2 = 0.
+        for axis in range(n_axes):
+            dist = abs(coords["particle1"][axis] -
+                       coords["particle2"][axis])
+            if dist > size_box[axis] * 0.5:
+                dist -= size_box[axis]
+            dist2 += dist**2
+        return np.sqrt(dist2)
+
+    @classmethod
+    def calculate_contact_times(
+            cls, system, accumulator_pos, contact_threshold):
+        """
+        Calculate the contact times from the time series of the coordinates
+        stored in ``accumulator_ids1`` and ``accumulator_ids2``.
+
```

Comment:
Note to self: I need to fix this broken docs.

Reply:
My plan is to merge this ASAP. Do you need more time to review my patch?

Reply:
I quickly went through your patch and it look fine to me. The only two issues  I spotted  these two.  If you can fix the docs here then I think you can proceed to merge it on my side!